addons/project/models/project.py

```diff
@@ -1260,6 +1274,16 @@ def _compute_ancestor_id(self):
         for task in self:
             task.ancestor_id = task.parent_id.ancestor_id or task.parent_id
 
+    def _search_is_private(self, operator, value):
+        if not isinstance(value, bool):
+            raise ValueError(_('Value should be True or False (not %s)'), value)
+        if operator not in ['=', '!=']:
+            raise NotImplementedError(_('Operation should be = or != (not %s)'), value)
+        if (operator == '=' and value) or (operator == '!=' and not value):
+            return [('project_id', '=', False), ('parent_id', '=', False)]
+        else:
+            return ['|', ('project_id', '!=', False), ('parent_id', '!=', False)]
```

Comment:
Why using `parent_id` in the domain? What are the use cases you are trying to cover?
```suggestion
        if (operator == '=' and value) or (operator == '!=' and not value):
            return [('project_id', '=', False)]
        else:
            return [('project_id', '!=', False)]
```

Reply:
i just followed the logic of _compute_is_private
`task.is_private = not task.project_id and not task.parent_id` 

so to search i need to check parent_id and project_id no ?

Reply:
It is not necessary, because a task is private only if there is not project set on this task.

Reply:
Why didn't you changed the code here?

Reply:
i though it's just a question and a waited for your confirmation, fixed !